enhancements/multi-arch/multiarch-manager-operator.md

```diff
@@ -0,0 +1,703 @@
+  that workloads continue to deploy and run as expected given that the images
+  used for deploying workloads may not support the above said secondary
+  architecture
+- As a cluster administrator who wants to migrate from a single-arch to a
+  multi-arch compute cluster, I do not want to rebuild each container image for
+  all the architectures that are present in the cluster
+- As a user who wants to deploy workloads on a multi-arch compute cluster, I do
+  not want to modify every pod (or any resources owning pods) spec to include a
+  node affinity predicate to filter the nodes at scheduling time, based on the
+  compatible architectures
+- As an operator developer, I want to allow users to develop operators only
+  supporting a subset of architectures or a single architecture without
+  additional business logic changes or requirements around having to specify
+  any affinities themselves. This will not be encouraged but will not become an
+  impediment to rolling out a feature that can work on multi-arch compute
+  clusters
+
+### Goals
+- Provide a way to ensure that pods are scheduled on nodes running a CPU
+  architecture supported by all the images of their containers. If such a node
+  does not exist, the pod holds in the `Pending` phase giving the user a clear
+  indication that the node affinity constraint is not met.
+
+### Non-Goals
+- This feature does not aim to modify the Kubernetes scheduler
+- This feature does not "schedule" pods; it influences scheduling by patching
+  the affinity of gated pods[^2]
+- Modification of fields other than the node affinity is not supported.
+  Moreover, the controller logic will only consider the
+  `kubernetes.io/architecture` label
+
+## Proposal
+Kubernetes 1.27 introduced the scheduling readiness feature[^1]. It
+allows users and controllers to set scheduling gates in the
+`podSpec.schedulingGates` list at admission time. Gated pods allow mutating some
+of the pod spec fields before the pod is scheduled. In particular, the
+scheduler's default `Enqueue` plugin holds the pods before entering the
+`ActiveQ` queue if a scheduling gate exists in the `podSpec.schedulingGates`
+list. Until gates are in the list, the pod will expose its status as
+`SchedulingGated`. The controller responsible for a given scheduling gate should
+implement the business logic so that the pod can be considered ready to be
+scheduled. KEP-3838[^2] introduced the possibility of mutating the node affinity
+of pods while they are in `schedulingGated` status (i.e., their
+`podSpec.schedulingGates` list has at least one gate).
+
+The multiarch-manager-operator will leverage the scheduling readiness feature
+and the mutable pod scheduling directives to update the node affinity of pods
+with the set of architectures they support. When a pod is submitted to the API
+server, a mutating admission webhook intercepts the pod and adds a scheduling
+gate to the pod spec.
```

Comment:
Hey @Prashanth684 and @aleskandro from what I understand here, the Operator will have two components? MutatingAdmissionWebHook and the SchedulingGate? 

With a complicated application, my colleagues in research found the mutating webhook resulted in an unintended out-of-sync status where the changed Pod resource didn't checksum correctly and the AppController ended up reconciling the resource since it wasn't as intended.  As such, maybe there should be some short-circuit methodology or flag in the metadata to skip the mutation.

Thanks for clarifying. Paul

Reply:
ok. if you can send me the details offline, we can look into it.